### PR TITLE
Refactor build system to prepare for building multiple Postgres versions.

### PR DESCRIPTION
**Pull request by hlinnaka**
Rename "vendor/postgres" to "vendor/postgres-v14"

Change Postgres build and install directory paths to be version-specific:

- tmp_install/build -> pg_install/build/14
- tmp_install/* -> pg_install/14/*

And Makefile targets:

- "make postgres" -> "make postgres-v14"
- "make postgres-headers" -> "make postgres-v14-headers"
- etc.
    
Add Makefile aliases:

- "make postgres" to build "postgres-v14" and in future, "postgres-v15"
- similarly for "make postgres-headers"

### CODE REVIEW

**Comment by hlinnaka** [comment]
@lubennikovaav, have you thought how we would build v14 and v15 side-by-side? I'm thinking something like this

**Comment by hlinnaka** [comment]
Here's what it will look like when we add v15 to it: https://github.com/neondatabase/neon/commit/ead121be1d92bbe9aa85b651ba50afebe71fd93b.

Lots of stuff remains to do, of course, like building postgres_ffi for both versions, refactoring code to isolate version-specific stuff to separate modules, carrying a postgres version number to all places that need it. But this is how the build scripts could look like.

**Comment by hlinnaka** [comment]
rebased

File: README.md [hunk 1]
```diff
@@ -101,7 +102,7 @@ make -j`sysctl -n hw.logicalcpu`
 ```
 
 #### Dependency installation notes
-To run the `psql` client, install the `postgresql-client` package or modify `PATH` and `LD_LIBRARY_PATH` to include `tmp_install/bin` and `tmp_install/lib`, respectively.
+To run the `psql` client, install the `postgresql-client` package or modify `PATH` and `LD_LIBRARY_PATH` to include `pg_install/bin` and `pg_install/lib`, respectively.
```
Comment (lubennikovaav) [inline README.md:105]: Shouldn't it be `pg_unstall/v14/bin` and pg_unstall/v14/lib` instead?

**Review by lubennikovaav — APPROVED** [review]

**Comment by hlinnaka** [comment]
This was obsoleted by PR #2337, closing.

**Pull request closed without merging**